ngraph/test/preprocess.cpp

```diff
@@ -0,0 +1,189 @@
+// Copyright (C) 2018-2021 Intel Corporation
+// SPDX-License-Identifier: Apache-2.0
+//
+
+#include "gtest/gtest.h"
+#include "ngraph/ngraph.hpp"
+#include "ngraph/ops.hpp"
+#include "openvino/core/preprocess/pre_post_process.hpp"
+#include "util/all_close.hpp"
+#include "util/all_close_f.hpp"
+#include "util/test_tools.hpp"
+
+using namespace ov;
+using namespace ov::preprocess;
+using namespace ngraph::test;
+
+static std::shared_ptr<Function> create_simple_function(element::Type type, const PartialShape& shape) {
+    auto data1 = std::make_shared<op::v0::Parameter>(type, shape);
+    data1->set_friendly_name("input1");
+    auto res = std::make_shared<op::v0::Result>(data1);
+    res->set_friendly_name("Result");
+    return std::make_shared<Function>(ResultVector{res}, ParameterVector{data1});
+}
+
+static std::shared_ptr<Function> create_2inputs(element::Type type, const PartialShape& shape) {
+    auto data1 = std::make_shared<op::v0::Parameter>(type, shape);
+    data1->set_friendly_name("input1");
+    auto data2 = std::make_shared<op::v0::Parameter>(type, shape);
+    data2->set_friendly_name("input2");
+    auto res1 = std::make_shared<op::v0::Result>(data1);
+    res1->set_friendly_name("Result");
+    auto res2 = std::make_shared<op::v0::Result>(data2);
+    res2->set_friendly_name("Result");
+    return std::make_shared<Function>(ResultVector{res1, res2}, ParameterVector{data1, data2});
+}
+
+TEST(pre_post_process, simple_mean_scale) {
+    auto f = create_simple_function(element::f32, ngraph::Shape{1, 3, 2, 2});
+    f = PrePostProcessor().input(InputInfo().preprocess(PreProcessSteps().mean(1.f).scale(2.f))).build(f);
```

Comment:
should not we also compare with reference ngraph function? like we do in transformations tests

Reply:
Comparing ngraph-functions relies on internal 'mean/scale' preprocess implementation. Currently simple Subtract & Divide are added, but later it can be changed to 'Add/Multiply'. And then even some 'smart' implementation can transform it automatically to 'Multiply-Add'.
In this case I wanted to test that 'mean' and 'scale' do work correctly from user's perspective. So if any such internal change is made, there will be no need to update the test case